crates/nix_rs/src/flake/metadata.rs

```diff
@@ -0,0 +1,19 @@
+use crate::command::{NixCmd, NixCmdError};
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct FlakeMetadata {
+    pub path: String,
```

Comment:
We should try not use `str`/ `String` for everything. When there's an appropriate type -- and there's one for paths -- we should try to use that.